include/zephyr/dmm/dmm.h

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright (c) 2023 Carlo Caione <ccaione@baylibre.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_DMM_DMM_H_
+#define ZEPHYR_INCLUDE_DMM_DMM_H_
+
+#include <stdio.h>
+#include <zephyr/device.h>
+#include <zephyr/kernel.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** @brief Get a DMA handle for outbound data transfer
+ *
+ *  This function is used to translate the user-provided buffer into something
+ *  that is usable for a DMA outbound data transfer and return that to the user
+ *  as an opaque DMA handle (usually a physical/bus address).
+ *
+ *  This function should:
+ *
+ *  - allocate bounce buffers if the user-provided buffer cannot be used
+ *    directly by the DMA (for example the memory is not DMA accessible, not
+ *    aligned, not padded to cache line size, etc...)
+ *  - copy the data from the user-provided buffer to the bounce buffers
+ *  - translate CPU buffer addresses to DMA addresses if needed
+ *  - do the proper cache management if buffers are allocated in cacheable
+ *    memory (for example flushing cache lines associated with the memory
+ *    backing the DMA accessible buffers)
+ *
+ *  @param[in] dev Device pointer
+ *  @param[in] flags Flags
+ *  @param[out] dma_handle DMA handle.
+ *  @param[in] buffer User-provided buffer.
+ *  @param[in] buffer_size Size of the user-provided buffer.
+ *
+ *  @retval 0 on success
+ *  @retval other errno codes depending on the implementation
+ */
+int dmm_get_buffer_for_dma_out(const struct device *dev, uint32_t flags,
```

Comment:
would this possible to define where is the buffer comes from? say a memory pool point?or just a fixed physical address